src/algorithms/TambyVanderpooten.jl

```diff
@@ -0,0 +1,216 @@
+    MOI.set(model.inner, MOI.ObjectiveSense(), MOI.MIN_SENSE)
+    U_N = Dict{Vector{Float64},Vector{Vector{Vector{Float64}}}}()
+    V = [Tuple{Vector{Float64},Vector{Float64}}[] for k in 1:n]
+    U_N[yN] = [[_get_child(yN, yI, k)] for k in 1:n]
+    status = MOI.OPTIMAL
+    while !isempty(U_N)
+        if _time_limit_exceeded(model, start_time)
+            status = MOI.TIME_LIMIT
+            break
+        end
+        k, u = _select_search_zone(U_N, yI)
+        MOI.set(
+            model.inner,
+            MOI.ObjectiveFunction{typeof(scalars[k])}(),
+            scalars[k],
+        )
+        ε_constraints = Any[]
+        for (i, f_i) in enumerate(scalars)
+            if i != k
+                ci = MOI.add_constraint(
+                    model.inner,
+                    f_i,
+                    MOI.LessThan{Float64}(u[i] - 1),
+                )
+                push!(ε_constraints, ci)
+            end
+        end
+        if u[k] ≠ yN[k]
+            if warm_start_supported
+                variables_start = solutions[first(U_N[u][k])]
+                for x_i in variables
+                    MOI.set(
+                        model.inner,
+                        MOI.VariablePrimalStart(),
+                        x_i,
+                        variables_start[x_i],
+                    )
+                end
+            end
+        end
+        MOI.optimize!(model.inner)
+        if !_is_scalar_status_optimal(model)
+            return status, nothing
+        end
+        y_k = MOI.get(model.inner, MOI.ObjectiveValue())
+        sum_f = sum(1.0 * s for s in scalars)
+        MOI.set(model.inner, MOI.ObjectiveFunction{typeof(sum_f)}(), sum_f)
+        y_k_constraint =
+            MOI.add_constraint(model.inner, scalars[k], MOI.EqualTo(y_k))
+        MOI.optimize!(model.inner)
+        if !_is_scalar_status_optimal(model)
+            return status, nothing
+        end
+        X, Y = _compute_point(model, variables, model.f)
+        MOI.delete.(model, ε_constraints)
+        MOI.delete(model, y_k_constraint)
+        push!(V[k], (u, Y))
+        if Y ∉ U_N[u][k]
+            _update_search_region(U_N, Y, yN)
+            solutions[Y] = X
+        end
+        bounds_to_remove = Vector{Float64}[]
+        for u_i in keys(U_N)
+            for k in 1:n
+                if u_i[k] == yI[k]
+                    push!(bounds_to_remove, u_i)
+                else
+                    for (u_j, y_j) in V[k]
+                        if all(_project(u_i, k) .<= _project(u_j, k)) &&
+                           (y_j[k] == u_i[k])
+                            push!(bounds_to_remove, u_i)
+                        end
+                    end
+                end
+            end
+        end
+        if !isempty(bounds_to_remove)
+            for bound_to_remove in bounds_to_remove
+                delete!(U_N, bound_to_remove)
+            end
+        end
+    end
+    solutions = [SolutionPoint(X, Y) for (Y, X) in solutions]
```

Comment:
The solutions look like they're in different orders on Julia 1 and Julia 1.6. You could sort them here to fix?

```suggestion
    solutions = [SolutionPoint(X, Y) for (Y, X) in solutions]
    sort!(solutions, by = s -> s.y)
```

Reply:
I did this but should this be here or in the tests?

Reply:
I think here, but you'll need to update the tests as well.